scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,64 @@
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str
```

Comment:
There are more test cases for parser in https://github.com/scrapy/scrapy/blob/master/tests/test_downloadermiddleware_robotstxt.py, can you copy them here
https://github.com/scrapy/scrapy/blob/f755507d682ba4e093c5ea85c6a4761d94c71436/tests/test_downloadermiddleware_robotstxt.py#L54
https://github.com/scrapy/scrapy/blob/f755507d682ba4e093c5ea85c6a4761d94c71436/tests/test_downloadermiddleware_robotstxt.py#L90
https://github.com/scrapy/scrapy/blob/f755507d682ba4e093c5ea85c6a4761d94c71436/tests/test_downloadermiddleware_robotstxt.py#L112